src/web/src/components/Banner.tsx

```diff
@@ -148,6 +150,35 @@ export default function Banner() {
     getGitData();
   }, []);
 
+  // Observer banner
+  useEffect(() => {
+    const options = {
+      root: null,
+      threshold: 0.9,
+    };
+
+    const observer = new IntersectionObserver(
+      (entries) =>
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            onVisibilityChange(true);
+          }
+          if (!entry.isIntersecting) {
+            onVisibilityChange(false);
+          }
+        }),
+      options
+    );
+    observer.observe(timelineAnchor.current!);
+
+    const timelineAnchorCopy = timelineAnchor.current;
+
+    return () => {
+      observer.unobserve(timelineAnchorCopy as HTMLDivElement);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is this necessary, or can you add `onVisibilityChange` to it?  Fine if you have to do it this way, just curious.